completion/waffle.py

```diff
@@ -4,26 +4,30 @@
 """
 
 
-try:
-    from openedx.core.djangoapps.waffle_utils import WaffleSwitchNamespace
-except ImportError:
-    pass
+from edx_toggles.toggles import WaffleSwitch, WaffleSwitchNamespace
 
 # Namespace
-WAFFLE_NAMESPACE = 'completion'
-
-# Switches
-
-# Full name: completion.enable_completion_tracking
-# Indicates whether or not to track completion of individual blocks.  Keeping
-# this disabled will prevent creation of BlockCompletion objects in the
-# database, as well as preventing completion-related network access by certain
-# xblocks.
-ENABLE_COMPLETION_TRACKING = 'enable_completion_tracking'
+WAFFLE_NAMESPACE = "completion"
 
 
 def waffle():
     """
     Returns the namespaced, cached, audited Waffle class for completion.
     """
-    return WaffleSwitchNamespace(name=WAFFLE_NAMESPACE, log_prefix='completion: ')
+    return WaffleSwitchNamespace(name=WAFFLE_NAMESPACE, log_prefix="completion: ")
+
+
+# Switches
+
+# The switch name variable is preserved for backward compatibility
+ENABLE_COMPLETION_TRACKING = "enable_completion_tracking"
+# .. toggle_name: completion.enable_completion_tracking
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: Indicates whether or not to track completion of individual blocks. Keeping this disabled
+#   will prevent creation of BlockCompletion objects in the database, as well as preventing completion-related
+#   network access by certain xblocks.
+# .. toggle_use_cases: open_edx
```

Comment:
@regisb: Do you know that this is meant to be a permanent flag, or was this just a guess?

Reply:
My bet is that this is meant to be a platform-togglable flag, because I was bit by this in Tutor: https://discuss.overhang.io/t/course-not-starting/962/12 I assumed that the fact that the flag was not enabled by default was a feature, and not a bug. Also, this flag has been there for a very long time, so even if the flag was meant to be temporary, it's now permanent.

Reply:
Sounds good.  I'm not clear on why it is an option and why it is off by default.  I asked the team and we'll see.
Either way, we can merge and it can always be updated later.